infrastructure/kubernetes/default/ingress.yml

```diff
@@ -0,0 +1,154 @@
+#
+# The purpose of the ingress controller is to route HTTP/HTTPS requests
+# on each cluster node to a specific service based on rules (see below).
+# We basically use this to expose all our apps and services via a single
+# HTTP/HTTPS endpoint.
+#
+# Documentation on the nginx ingress controller:
+# https://github.com/kubernetes/ingress-nginx
+#
+---
+#
+# A default backend is required to serve requests that don't get catched by
+# the rules configured in the ingress controller
+#
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: default-http-backend
+  labels:
+    app: default-http-backend
+  namespace: default
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: default-http-backend
+    spec:
+      terminationGracePeriodSeconds: 60
+      containers:
+      - name: default-http-backend
+        # Any image is permissable as long as:
+        # 1. It serves a 404 page at /
+        # 2. It serves 200 on a /healthz endpoint
+        image: gcr.io/google_containers/defaultbackend:1.4
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8080
+            scheme: HTTP
+          initialDelaySeconds: 30
+          timeoutSeconds: 5
+        ports:
+        - containerPort: 8080
+        resources:
+          limits:
+            cpu: 10m
+            memory: 20Mi
+          requests:
+            cpu: 10m
+            memory: 20Mi
+---
+#
+# Expose the default backend pod as a service
+#
+apiVersion: v1
+kind: Service
+metadata:
+  name: default-http-backend
+  namespace: default
+  labels:
+    app: default-http-backend
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+  selector:
+    app: default-http-backend
+
+---
+#
+# The ingress controller dispatches HTTP requests to service pods
+#
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: nginx-ingress-controller
```

Comment:
TODO we may even use [linkerd](https://buoyant.io/2017/04/06/a-service-mesh-for-kubernetes-part-viii-linkerd-as-an-ingress-controller/) as an ingress, not sure whether it makes sense since our service mesh is really simple